### PR TITLE
Added the SomeApply wart

### PR DESCRIPTION
**Pull request by omervk**
Submitting the same pull request described here: https://github.com/wartremover/wartremover/pull/293

### CODE REVIEW

**Comment by amanjpro** [comment]
Thank you for the patch, will review it later today when I have time

File: src/test/scala/wartremover/contrib/warts/SomeApplyTest.scala [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  }
+  test("can use Some.unapply in pattern matching") {
+    val result = WartTestTraverser(SomeApply) {
+      Option("test") match {
+        case Some(test) => println(test)
+        case None => println("not gonna happen")
+      }
+    }
+    assertEmpty(result)
+  }
```
Comment (amanjpro) [inline src/test/scala/wartremover/contrib/warts/SomeApplyTest.scala]: Can you also add a test case for suppressed expressions? something like:

```scala
@SuppressWarnings("....")
Some(null) ...
```

**Comment by omervk** [comment]
I've added the relevant test and found that the directory structure does not match the package structure, which caused me to place the `SomeApply` wart under `wartremover.contrib.warts` instead of `org.wartremover.contrib.warts`. I recommend changing the directory structure to match the package structure to prevent future issues.

**Comment by amanjpro** [comment]
Thank you again @omervk 

@ChrisNeveu how about we move all the classes to src/warts? After all packages in Scala not necessarily match with directory structure

**Comment by ChrisNeveu** [comment]
I just set up the package structure to mirror Wartremover but feel free to reorganize. I recommend mergin this PR first though if you do :)

**Comment by amanjpro** [comment]
Good. I will merge this and then later tonight reorganize the directories

**Pull request merged**